RollingUpdateTest.shouldPerformRollingUpgrade flaky

**Summary**

- How often does the test fail? several times since the last fix for this test (new and different error message)
- Does it block your work? no
- Do we suspect that it is a real failure? unknown

**Failures**

<details><summary>Example assertion failure</summary>
<pre>

org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:330)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
	at io.zeebe.test.RollingUpdateTest.shouldPerformRollingUpgrade(RollingUpdateTest.java:144)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	... 43 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:498)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 44 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for URL to be accessible (http://localhost:32825/ready should return HTTP [204])
	at org.testcontainers.containers.wait.strategy.HttpWaitStrategy.waitUntilReady(HttpWaitStrategy.java:214)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:35)
	at io.zeebe.containers.BrokerWaitStrategy.waitUntilReady(BrokerWaitStrategy.java:45)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:890)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:441)
	... 46 more

</pre>
</details>

**Hypotheses**


**Logs**

<details><summary>Logs</summary>
<pre>
00:39:17.181 [main] [] INFO  io.zeebe.test - Test started: shouldPerformRollingUpgrade(io.zeebe.test.RollingUpdateTest)
00:39:26.386 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy - Found docker client settings from environment
00:39:26.404 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=tcp://localhost:2375
00:39:28.205 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.DockerClientFactory - Docker host IP address is localhost
00:39:28.305 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.DockerClientFactory - Connected to docker: 
  Server Version: 18.09.4
  API Version: 1.39
  Operating System: Alpine Linux v3.9 (containerized)
  Total Memory: 120871 MB
00:39:28.611 [testcontainers-lifecycle-1] [] WARN  org.testcontainers.utility.RegistryAuthLocator - Failure when attempting to lookup auth config (dockerImageName: testcontainersofficial/ryuk:0.3.0, configFile: /root/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /root/.docker/config.json (No such file or directory)
00:39:30.090 [tc-okhttp-stream-1239130408] [] INFO  org.testcontainers.DockerClientFactory - Starting to pull image
00:39:30.192 [tc-okhttp-stream-1239130408] [] INFO  org.testcontainers.DockerClientFactory - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
00:39:33.485 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.DockerClientFactory - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
00:39:33.486 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.DockerClientFactory - Checking the system...
00:39:33.493 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
00:39:33.683 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker environment should have more than 2GB free disk space
00:39:33.706 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling docker image: camunda/zeebe:0.23.0. Please be patient; this may take some time but only needs to be done once.
00:39:33.706 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling docker image: camunda/zeebe:0.23.0. Please be patient; this may take some time but only needs to be done once.
00:39:33.707 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling docker image: camunda/zeebe:0.23.0. Please be patient; this may take some time but only needs to be done once.
00:39:34.978 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting to pull image
00:39:34.979 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
00:39:34.996 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting to pull image
00:39:34.998 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
00:39:34.998 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting to pull image
00:39:35.002 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
00:39:35.499 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 11 pending,  1 downloaded,  0 extracted, (673 KB/? MB)
00:39:35.502 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 11 pending,  1 downloaded,  0 extracted, (673 KB/? MB)
00:39:35.510 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 11 pending,  1 downloaded,  0 extracted, (673 KB/? MB)
00:39:35.515 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 10 pending,  2 downloaded,  0 extracted, (20 MB/? MB)
00:39:35.516 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 10 pending,  2 downloaded,  0 extracted, (20 MB/? MB)
00:39:35.518 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 10 pending,  2 downloaded,  0 extracted, (20 MB/? MB)
00:39:35.878 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  9 pending,  3 downloaded,  0 extracted, (54 MB/? MB)
00:39:35.880 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  9 pending,  3 downloaded,  0 extracted, (54 MB/? MB)
00:39:35.881 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  9 pending,  3 downloaded,  0 extracted, (54 MB/? MB)
00:39:35.940 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  8 pending,  4 downloaded,  0 extracted, (54 MB/? MB)
00:39:35.940 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  8 pending,  4 downloaded,  0 extracted, (54 MB/? MB)
00:39:35.978 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  8 pending,  4 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.020 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  7 pending,  5 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.017 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  7 pending,  5 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.017 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  7 pending,  5 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.379 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  6 pending,  6 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.379 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  6 pending,  6 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.380 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  6 pending,  6 downloaded,  0 extracted, (54 MB/? MB)
00:39:36.479 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  5 pending,  7 downloaded,  0 extracted, (64 MB/? MB)
00:39:36.480 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  5 pending,  7 downloaded,  0 extracted, (64 MB/? MB)
00:39:36.484 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  5 pending,  7 downloaded,  0 extracted, (64 MB/? MB)
00:39:36.787 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  4 pending,  8 downloaded,  0 extracted, (87 MB/? MB)
00:39:36.788 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  4 pending,  8 downloaded,  0 extracted, (87 MB/? MB)
00:39:36.801 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  4 pending,  8 downloaded,  0 extracted, (87 MB/? MB)
00:39:37.279 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  3 pending,  9 downloaded,  0 extracted, (121 MB/? MB)
00:39:37.280 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  3 pending,  9 downloaded,  0 extracted, (121 MB/? MB)
00:39:37.281 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  3 pending,  9 downloaded,  0 extracted, (121 MB/? MB)
00:39:37.778 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  2 pending, 10 downloaded,  0 extracted, (190 MB/? MB)
00:39:37.779 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  2 pending, 10 downloaded,  0 extracted, (190 MB/? MB)
00:39:37.780 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  2 pending, 10 downloaded,  0 extracted, (190 MB/? MB)
00:39:38.835 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  1 pending, 11 downloaded,  0 extracted, (280 MB/? MB)
00:39:38.878 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  0 extracted, (280 MB/306 MB)
00:39:38.878 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  1 pending, 11 downloaded,  0 extracted, (280 MB/? MB)
00:39:38.879 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  0 extracted, (280 MB/306 MB)
00:39:38.879 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  1 pending, 11 downloaded,  0 extracted, (280 MB/? MB)
00:39:38.880 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  0 extracted, (280 MB/306 MB)
00:39:39.208 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  1 extracted, (280 MB/306 MB)
00:39:39.211 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  1 extracted, (280 MB/306 MB)
00:39:39.211 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  1 extracted, (280 MB/306 MB)
00:39:39.978 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  2 extracted, (288 MB/306 MB)
00:39:39.981 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  2 extracted, (288 MB/306 MB)
00:39:39.985 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  2 extracted, (288 MB/306 MB)
00:39:40.806 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  3 extracted, (289 MB/306 MB)
00:39:40.806 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  3 extracted, (289 MB/306 MB)
00:39:40.810 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  3 extracted, (289 MB/306 MB)
00:39:41.378 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  4 extracted, (294 MB/306 MB)
00:39:41.378 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  4 extracted, (294 MB/306 MB)
00:39:41.378 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  4 extracted, (294 MB/306 MB)
00:39:41.593 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  5 extracted, (294 MB/306 MB)
00:39:41.593 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  5 extracted, (294 MB/306 MB)
00:39:41.596 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  5 extracted, (294 MB/306 MB)
00:39:45.096 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  6 extracted, (300 MB/306 MB)
00:39:45.099 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  6 extracted, (300 MB/306 MB)
00:39:45.099 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  6 extracted, (300 MB/306 MB)
00:39:45.339 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  7 extracted, (300 MB/306 MB)
00:39:45.379 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  7 extracted, (300 MB/306 MB)
00:39:45.386 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  7 extracted, (300 MB/306 MB)
00:39:45.691 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  8 extracted, (300 MB/306 MB)
00:39:45.691 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  8 extracted, (300 MB/306 MB)
00:39:45.704 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  8 extracted, (300 MB/306 MB)
00:39:49.278 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  9 extracted, (303 MB/306 MB)
00:39:49.279 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  9 extracted, (303 MB/306 MB)
00:39:49.279 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  9 extracted, (303 MB/306 MB)
00:39:52.678 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 10 extracted, (305 MB/306 MB)
00:39:52.678 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 10 extracted, (305 MB/306 MB)
00:39:52.681 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 10 extracted, (305 MB/306 MB)
00:39:52.805 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 11 extracted, (305 MB/306 MB)
00:39:52.805 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 11 extracted, (305 MB/306 MB)
00:39:52.805 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 11 extracted, (305 MB/306 MB)
00:39:52.928 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 12 extracted, (306 MB/306 MB)
00:39:52.978 [tc-okhttp-stream-738877513] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pull complete. 12 layers, pulled in 17s (downloaded 306 MB at 18 MB/s)
00:39:52.978 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 12 extracted, (306 MB/306 MB)
00:39:52.978 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 12 extracted, (306 MB/306 MB)
00:39:52.982 [tc-okhttp-stream-1145082700] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pull complete. 12 layers, pulled in 18s (downloaded 306 MB at 17 MB/s)
00:39:52.983 [tc-okhttp-stream-1917895787] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pull complete. 12 layers, pulled in 17s (downloaded 306 MB at 18 MB/s)
00:39:53.105 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Creating container for image: camunda/zeebe:0.23.0
00:39:53.106 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Creating container for image: camunda/zeebe:0.23.0
00:39:53.108 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Creating container for image: camunda/zeebe:0.23.0
00:40:19.714 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting container with ID: 05ada3e25f568b313b1dc28d9ba967b969e8ab341b9bce859c49571e5f70e15e
00:40:20.669 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting container with ID: a8c71c2e0b93d6146cd2ec2727050824486315091d2116b0ba2cc52a6f8b3348
00:40:20.732 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting container with ID: f9715b562ccc6498ed42c4e79542e89b02a91c2575ba22d6f90abf39ad8c307e
00:40:21.501 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 is starting: 05ada3e25f568b313b1dc28d9ba967b969e8ab341b9bce859c49571e5f70e15e
00:40:21.512 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-5j3p7C-gXjoha: Waiting for 60 seconds for URL: http://localhost:32783/ready
00:40:22.116 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 is starting: a8c71c2e0b93d6146cd2ec2727050824486315091d2116b0ba2cc52a6f8b3348
00:40:22.118 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 is starting: f9715b562ccc6498ed42c4e79542e89b02a91c2575ba22d6f90abf39ad8c307e
00:40:22.515 [testcontainers-lifecycle-0] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-KtxHtV-gc4hK9: Waiting for 60 seconds for URL: http://localhost:32788/ready
00:40:22.516 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-Xjsuje-mJ7EzN: Waiting for 60 seconds for URL: http://localhost:32791/ready
00:41:00.405 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 started in PT1M39.508816S
00:41:00.405 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 started in PT1M39.506756S
00:41:02.588 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 started in PT1M41.690506S
00:41:17.247 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Creating container for image: camunda/zeebe:current-test
00:41:17.364 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Starting container with ID: 1bea8b9360a2cbbcfd0345fac924769e0a92a555f8883757e4b4c5cf1471aacd
00:41:19.592 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test is starting: 1bea8b9360a2cbbcfd0345fac924769e0a92a555f8883757e4b4c5cf1471aacd
00:41:19.681 [main] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-dSczZw-SEgZdz: Waiting for 60 seconds for URL: http://localhost:32817/ready
00:41:43.706 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test started in PT26.459848S
00:41:51.737 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Creating container for image: camunda/zeebe:current-test
00:41:51.825 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Starting container with ID: 46f1aad2492b7a8cb377a33d4dc3a855719f46ff89ce59531654edc102b7ba13
00:41:52.415 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test is starting: 46f1aad2492b7a8cb377a33d4dc3a855719f46ff89ce59531654edc102b7ba13
00:41:52.423 [main] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-fo6WB2-Cd9qy2: Waiting for 60 seconds for URL: http://localhost:32825/ready
00:42:52.427 [main] [] ERROR 🐳 [camunda/zeebe:current-test] - Could not start container
org.testcontainers.containers.ContainerLaunchException: Timed out waiting for URL to be accessible (http://localhost:32825/ready should return HTTP [204])
	at org.testcontainers.containers.wait.strategy.HttpWaitStrategy.waitUntilReady(HttpWaitStrategy.java:214) ~[testcontainers-1.14.3.jar:?]
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:35) ~[testcontainers-1.14.3.jar:?]
	at io.zeebe.containers.BrokerWaitStrategy.waitUntilReady(BrokerWaitStrategy.java:45) ~[zeebe-test-container-0.33.0.jar:0.33.0]
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:890) ~[testcontainers-1.14.3.jar:?]
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:441) ~[testcontainers-1.14.3.jar:?]
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325) ~[testcontainers-1.14.3.jar:?]
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81) [duct-tape-1.0.8.jar:?]
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323) [testcontainers-1.14.3.jar:?]
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311) [testcontainers-1.14.3.jar:?]
	at io.zeebe.test.RollingUpdateTest.shouldPerformRollingUpgrade(RollingUpdateTest.java:144) [test-classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) [junit-4.13.jar:4.13]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.13.jar:4.13]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) [junit-4.13.jar:4.13]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit-4.13.jar:4.13]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit-4.13.jar:4.13]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) [junit-4.13.jar:4.13]
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) [junit-4.13.jar:4.13]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) [junit-4.13.jar:4.13]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413) [junit-4.13.jar:4.13]
	at org.junit.runners.Suite.runChild(Suite.java:128) [junit-4.13.jar:4.13]
	at org.junit.runners.Suite.runChild(Suite.java:27) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) [junit-4.13.jar:4.13]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413) [junit-4.13.jar:4.13]
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55) [surefire-junit47-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137) [surefire-junit47-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119) [surefire-junit47-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87) [surefire-junit47-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75) [surefire-junit47-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157) [surefire-junit47-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428) [surefire-booter-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162) [surefire-booter-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562) [surefire-booter-3.0.0-M5.jar:3.0.0-M5]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548) [surefire-booter-3.0.0-M5.jar:3.0.0-M5]
00:42:52.462 [main] [] ERROR 🐳 [camunda/zeebe:current-test] - Log output from the failed container:
+ export ZEEBE_HOST=zeebe-broker-fo6WB2

+ ZEEBE_HOST=zeebe-broker-fo6WB2

+ '[' false = true ']'

+ export ZEEBE_BROKER_NETWORK_HOST=0.0.0.0

+ ZEEBE_BROKER_NETWORK_HOST=0.0.0.0

+ export ZEEBE_BROKER_GATEWAY_CLUSTER_HOST=0.0.0.0

+ ZEEBE_BROKER_GATEWAY_CLUSTER_HOST=0.0.0.0

+ exec /usr/local/zeebe/bin/broker

2020-07-22 00:41:54.383 [] [main] WARN  io.zeebe.legacy - Found use of legacy system environment variable 'ZEEBE_EMBED_GATEWAY'. Please use 'ZEEBE_BROKER_GATEWAY_ENABLE' instead.

2020-07-22 00:41:54.391 [] [main] INFO  io.zeebe.legacy - The old environment variable is currently supported as part of our backwards compatibility goals.

2020-07-22 00:41:54.392 [] [main] INFO  io.zeebe.legacy - However, please note that support for the old environment variable is scheduled to be removed for release 0.24.0.

2020-07-22 00:41:54.393 [] [main] WARN  io.zeebe.legacy - Found use of legacy system environment variable 'ZEEBE_PARTITIONS_COUNT'. Please use 'ZEEBE_BROKER_CLUSTER_PARTITIONSCOUNT' instead.

2020-07-22 00:41:54.393 [] [main] INFO  io.zeebe.legacy - The old environment variable is currently supported as part of our backwards compatibility goals.

2020-07-22 00:41:54.394 [] [main] INFO  io.zeebe.legacy - However, please note that support for the old environment variable is scheduled to be removed for release 0.24.0.

  ______  ______   ______   ____    ______     ____    _____     ____    _  __  ______   _____  

 |___  / |  ____| |  ____| |  _ \  |  ____|   |  _ \  |  __ \   / __ \  | |/ / |  ____| |  __ \ 

    / /  | |__    | |__    | |_) | | |__      | |_) | | |__) | | |  | | | ' /  | |__    | |__) |

   / /   |  __|   |  __|   |  _ <  |  __|     |  _ <  |  _  /  | |  | | |  <   |  __|   |  _  / 

  / /__  | |____  | |____  | |_) | | |____    | |_) | | | \ \  | |__| | | . \  | |____  | | \ \ 

 /_____| |______| |______| |____/  |______|   |____/  |_|  \_\  \____/  |_|\_\ |______| |_|  \_\

                                                                                                

2020-07-22 00:41:55.214 [] [main] INFO  io.zeebe.broker.StandaloneBroker - Starting StandaloneBroker v0.25.0-SNAPSHOT on 46f1aad2492b with PID 6 (/usr/local/zeebe/lib/zeebe-distribution-0.25.0-SNAPSHOT.jar started by root in /usr/local/zeebe)

2020-07-22 00:41:55.220 [] [main] DEBUG io.zeebe.broker.StandaloneBroker - Running with Spring Boot v2.3.1.RELEASE, Spring v5.2.7.RELEASE

2020-07-22 00:41:55.222 [] [main] INFO  io.zeebe.broker.StandaloneBroker - No active profile set, falling back to default profiles: default

2020-07-22 00:41:57.728 [] [main] INFO  org.springframework.boot.web.embedded.tomcat.TomcatWebServer - Tomcat initialized with port(s): 9600 (http)

2020-07-22 00:41:57.791 [] [main] INFO  org.apache.coyote.http11.Http11NioProtocol - Initializing ProtocolHandler ["http-nio-0.0.0.0-9600"]

2020-07-22 00:41:57.793 [] [main] INFO  org.apache.catalina.core.StandardService - Starting service [Tomcat]

2020-07-22 00:41:57.795 [] [main] INFO  org.apache.catalina.core.StandardEngine - Starting Servlet engine: [Apache Tomcat/9.0.37]

2020-07-22 00:41:58.044 [] [main] INFO  org.apache.catalina.core.ContainerBase.[Tomcat].[localhost].[/] - Initializing Spring embedded WebApplicationContext

2020-07-22 00:41:58.045 [] [main] INFO  org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext - Root WebApplicationContext: initialization completed in 2771 ms

2020-07-22 00:41:59.005 [] [main] INFO  org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor - Initializing ExecutorService 'applicationTaskExecutor'

2020-07-22 00:41:59.885 [] [main] INFO  org.springframework.boot.actuate.endpoint.web.EndpointLinksResolver - Exposing 2 endpoint(s) beneath base path '/actuator'

2020-07-22 00:41:59.987 [] [main] INFO  org.apache.coyote.http11.Http11NioProtocol - Starting ProtocolHandler ["http-nio-0.0.0.0-9600"]

2020-07-22 00:42:00.114 [] [main] INFO  org.springframework.boot.web.embedded.tomcat.TomcatWebServer - Tomcat started on port(s): 9600 (http) with context path ''

2020-07-22 00:42:00.194 [] [main] INFO  io.zeebe.broker.StandaloneBroker - Started StandaloneBroker in 5.656 seconds (JVM running for 7.782)

2020-07-22 00:42:00.233 [] [main] DEBUG io.zeebe.broker.system - Initializing system with base path /usr/local/zeebe

2020-07-22 00:42:00.318 [] [main] INFO  io.zeebe.broker.system - Version: 0.25.0-SNAPSHOT

2020-07-22 00:42:00.711 [] [main] INFO  io.zeebe.broker.system - Starting broker 1 with configuration {

  "network" : {

    "host" : "0.0.0.0",

    "portOffset" : 0,

    "maxMessageSize" : "0MB",

    "advertisedHost" : "broker-1",

    "commandApi" : {

      "host" : "0.0.0.0",

      "port" : 26501,

      "advertisedHost" : "broker-1",

      "advertisedPort" : 26501,

      "address" : "0.0.0.0:26501",

      "advertisedAddress" : "broker-1:26501"

    },

    "internalApi" : {

      "host" : "0.0.0.0",

      "port" : 26502,

      "advertisedHost" : "broker-1",

      "advertisedPort" : 26502,

      "address" : "0.0.0.0:26502",

      "advertisedAddress" : "broker-1:26502"

    },

    "monitoringApi" : {

      "host" : "0.0.0.0",

      "port" : 9600,

      "advertisedHost" : "broker-1",

      "advertisedPort" : 9600,

      "address" : "0.0.0.0:9600",

      "advertisedAddress" : "broker-1:9600"

    },

    "maxMessageSizeInBytes" : 131072

  },

  "cluster" : {

    "initialContactPoints" : [ "broker-0:26502", "broker-1:26502", "broker-2:26502" ],

    "partitionIds" : [ 1 ],

    "nodeId" : 1,

    "partitionsCount" : 1,

    "replicationFactor" : 3,

    "clusterSize" : 3,

    "clusterName" : "zeebe-cluster",

    "membership" : {

      "broadcastUpdates" : false,

      "broadcastDisputes" : true,

      "notifySuspect" : false,

      "gossipInterval" : "PT0.25S",

      "gossipFanout" : 2,

      "probeInterval" : "PT1S",

      "probeTimeout" : "PT2S",

      "suspectProbes" : 3,

      "failureTimeout" : "PT10S",

      "syncInterval" : "PT10S"

    }

  },

  "threads" : {

    "cpuThreadCount" : 2,

    "ioThreadCount" : 2

  },

  "data" : {

    "directories" : [ "/usr/local/zeebe/data" ],

    "logSegmentSize" : "512MB",

    "snapshotPeriod" : "PT15M",

    "logIndexDensity" : 100,

    "diskUsageReplicationWatermark" : 0.9,

    "diskUsageCommandWatermark" : 0.8,

    "diskUsageMonitoringInterval" : "PT1S",

    "freeDiskSpaceCommandWatermark" : 20248258150,

    "freeDiskSpaceReplicationWatermark" : 10124129075,

    "logSegmentSizeInBytes" : 536870912,

    "atomixStorageLevel" : "DISK"

  },

  "exporters" : {

    "DebugLogExporter" : {

      "jarPath" : null,

      "className" : "io.zeebe.broker.exporter.debug.DebugLogExporter",

      "args" : {

        "prettyPrint" : false

      },

      "external" : false

    }

  },

  "gateway" : {

    "network" : {

      "host" : "0.0.0.0",

      "port" : 26500,

      "minKeepAliveInterval" : "PT30S"

    },

    "cluster" : {

      "contactPoint" : "0.0.0.0:26502",

      "requestTimeout" : "PT15S",

      "clusterName" : "zeebe-cluster",

      "memberId" : "gateway",

      "host" : "0.0.0.0",

      "port" : 26502,

      "membership" : {

        "broadcastUpdates" : false,

        "broadcastDisputes" : true,

        "notifySuspect" : false,

        "gossipInterval" : "PT0.25S",

        "gossipFanout" : 2,

        "probeInterval" : "PT1S",

        "probeTimeout" : "PT2S",

        "suspectProbes" : 3,

        "failureTimeout" : "PT10S",

        "syncInterval" : "PT10S"

      }

    },

    "threads" : {

      "managementThreads" : 1

    },

    "monitoring" : {

      "enabled" : false,

      "host" : "0.0.0.0",

      "port" : 9600

    },

    "security" : {

      "enabled" : false,

      "certificateChainPath" : null,

      "privateKeyPath" : null

    },

    "longPolling" : {

      "enabled" : true

    },

    "initialized" : true,

    "enable" : true

  },

  "backpressure" : {

    "enabled" : true,

    "algorithm" : "VEGAS",

    "aimd" : {

      "requestTimeout" : "PT1S",

      "initialLimit" : 100,

      "minLimit" : 1,

      "maxLimit" : 1000,

      "backoffRatio" : 0.9

    },

    "fixedLimit" : {

      "limit" : 20

    },

    "vegas" : {

      "alpha" : 3,

      "beta" : 6,

      "initialLimit" : 20

    },

    "gradient" : {

      "minLimit" : 10,

      "initialLimit" : 20,

      "rttTolerance" : 2.0

    },

    "gradient2" : {

      "minLimit" : 10,

      "initialLimit" : 20,

      "rttTolerance" : 2.0,

      "longWindow" : 600

    }

  },

  "stepTimeout" : "PT5M",

  "executionMetricsExporterEnabled" : false

}

2020-07-22 00:42:00.783 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [1/13]: actor scheduler

2020-07-22 00:42:00.828 [] [main] DEBUG io.zeebe.broker.system - Bootstrap Broker-1 [1/13]: actor scheduler started in 44 ms

2020-07-22 00:42:00.829 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [2/13]: membership and replication protocol

2020-07-22 00:42:00.832 [] [http-nio-0.0.0.0-9600-exec-1] INFO  org.apache.catalina.core.ContainerBase.[Tomcat].[localhost].[/] - Initializing Spring DispatcherServlet 'dispatcherServlet'

2020-07-22 00:42:00.833 [] [http-nio-0.0.0.0-9600-exec-1] INFO  org.springframework.web.servlet.DispatcherServlet - Initializing Servlet 'dispatcherServlet'

2020-07-22 00:42:00.845 [] [main] DEBUG io.zeebe.broker.clustering - Member 1 will contact node: broker-0:26502

2020-07-22 00:42:00.880 [] [main] DEBUG io.zeebe.broker.clustering - Member 1 will contact node: broker-1:26502

2020-07-22 00:42:00.881 [] [http-nio-0.0.0.0-9600-exec-1] INFO  org.springframework.web.servlet.DispatcherServlet - Completed initialization in 48 ms

2020-07-22 00:42:00.881 [] [main] DEBUG io.zeebe.broker.clustering - Member 1 will contact node: broker-2:26502

2020-07-22 00:42:05.594 [] [main] DEBUG io.zeebe.broker.system - Bootstrap Broker-1 [2/13]: membership and replication protocol started in 4765 ms

2020-07-22 00:42:05.595 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [3/13]: command api transport

2020-07-22 00:42:06.038 [] [main] DEBUG io.zeebe.broker.system - Bound command API to broker-1:26501 

2020-07-22 00:42:06.089 [] [main] DEBUG io.zeebe.broker.system - Bootstrap Broker-1 [3/13]: command api transport started in 494 ms

2020-07-22 00:42:06.090 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [4/13]: command api handler

2020-07-22 00:42:06.186 [] [main] DEBUG io.zeebe.broker.system - Bootstrap Broker-1 [4/13]: command api handler started in 96 ms

2020-07-22 00:42:06.187 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [5/13]: subscription api

2020-07-22 00:42:06.224 [] [main] DEBUG io.zeebe.broker.system - Bootstrap Broker-1 [5/13]: subscription api started in 37 ms

2020-07-22 00:42:06.224 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [6/13]: embedded gateway

2020-07-22 00:42:06.235 [] [main] INFO  io.zeebe.gateway - Version: 0.25.0-SNAPSHOT

2020-07-22 00:42:06.236 [] [main] INFO  io.zeebe.gateway - Starting gateway with configuration {

  "network" : {

    "host" : "0.0.0.0",

    "port" : 26500,

    "minKeepAliveInterval" : "PT30S"

  },

  "cluster" : {

    "contactPoint" : "0.0.0.0:26502",

    "requestTimeout" : "PT15S",

    "clusterName" : "zeebe-cluster",

    "memberId" : "gateway",

    "host" : "0.0.0.0",

    "port" : 26502,

    "membership" : {

      "broadcastUpdates" : false,

      "broadcastDisputes" : true,

      "notifySuspect" : false,

      "gossipInterval" : "PT0.25S",

      "gossipFanout" : 2,

      "probeInterval" : "PT1S",

      "probeTimeout" : "PT2S",

      "suspectProbes" : 3,

      "failureTimeout" : "PT10S",

      "syncInterval" : "PT10S"

    }

  },

  "threads" : {

    "managementThreads" : 1

  },

  "monitoring" : {

    "enabled" : false,

    "host" : "0.0.0.0",

    "port" : 9600

  },

  "security" : {

    "enabled" : false,

    "certificateChainPath" : null,

    "privateKeyPath" : null

  },

  "longPolling" : {

    "enabled" : true

  },

  "initialized" : true,

  "enable" : true

}

2020-07-22 00:42:06.405 [] [main] DEBUG io.zeebe.broker.system - Bootstrap Broker-1 [6/13]: embedded gateway started in 179 ms

2020-07-22 00:42:06.405 [] [main] INFO  io.zeebe.broker.system - Bootstrap Broker-1 [7/13]: cluster services

2020-07-22 00:42:06.989 [GatewayTopologyManager] [Broker-1-zb-actors-1] DEBUG io.zeebe.gateway - Received new broker BrokerInfo{nodeId=0, partitionsCount=1, clusterSize=3, replicationFactor=3, partitionRoles={1=LEADER}, partitionLeaderTerms={1=1}, version=0.23.0}.

2020-07-22 00:42:07.403 [] [raft-server-1-raft-partition-partition-1] ERROR io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-1} - An uncaught exception occurred, transition to inactive role

java.lang.IndexOutOfBoundsException: index=9224 length=4 capacity=885

	at org.agrona.concurrent.UnsafeBuffer.boundsCheck0(UnsafeBuffer.java:1692) ~[agrona-1.6.0.jar:1.6.0]

	at org.agrona.concurrent.UnsafeBuffer.getInt(UnsafeBuffer.java:472) ~[agrona-1.6.0.jar:1.6.0]

	at io.atomix.raft.snapshot.impl.SnapshotChunkDecoder.snapshotId(SnapshotChunkDecoder.java:341) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at io.atomix.raft.snapshot.impl.SnapshotChunkImpl.wrap(SnapshotChunkImpl.java:102) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at io.atomix.raft.roles.PassiveRole.onInstall(PassiveRole.java:189) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at io.atomix.raft.roles.FollowerRole.onInstall(FollowerRole.java:88) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at io.atomix.raft.impl.RaftContext.lambda$registerHandlers$2(RaftContext.java:208) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at io.atomix.raft.impl.RaftContext.lambda$runOnContext$19(RaftContext.java:224) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at io.atomix.utils.concurrent.SingleThreadContext$WrappedRunnable.run(SingleThreadContext.java:188) [atomix-utils-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]

	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]

	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]

	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]

	at java.lang.Thread.run(Unknown Source) [?:?]

2020-07-22 00:42:07.556 [GatewayTopologyManager] [Broker-1-zb-actors-1] DEBUG io.zeebe.gateway - Received new broker BrokerInfo{nodeId=2, partitionsCount=1, clusterSize=3, replicationFactor=3, partitionRoles={1=FOLLOWER}, partitionLeaderTerms={}, version=0.25.0-SNAPSHOT}.
</pre>
</details>

--- a/atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java
+++ b/atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java
@@ -38,6 +38,8 @@ import io.atomix.raft.protocol.VoteResponse;
 import io.atomix.raft.snapshot.PersistedSnapshot;
 import io.atomix.raft.snapshot.PersistedSnapshotListener;
 import io.atomix.raft.snapshot.ReceivedSnapshot;
+import io.atomix.raft.snapshot.SnapshotChunk;
+import io.atomix.raft.snapshot.impl.LegacySnapshotChunk;
 import io.atomix.raft.snapshot.impl.SnapshotChunkImpl;
 import io.atomix.raft.storage.log.RaftLogReader;
 import io.atomix.raft.storage.log.RaftLogWriter;
@@ -164,14 +166,16 @@ public class PassiveRole extends InactiveRole {
     // If the snapshot already exists locally, do not overwrite it with a replicated snapshot.
     // Simply reply to the
     // request successfully.
-    final var optLatestSnapshot = raft.getPersistedSnapshotStore().getLatestSnapshot();
-    if (optLatestSnapshot.isPresent()) {
-      if (optLatestSnapshot.get().getIndex() >= request.index()) {
-        abortPendingSnapshots();
+    final var latestSnapshotIndex =
+        raft.getPersistedSnapshotStore()
+            .getLatestSnapshot()
+            .map(PersistedSnapshot::getIndex)
+            .orElse(Long.MIN_VALUE);
+    if (latestSnapshotIndex >= request.index()) {
+      abortPendingSnapshots();
 
-        return CompletableFuture.completedFuture(
-            logResponse(InstallResponse.builder().withStatus(RaftResponse.Status.OK).build()));
-      }
+      return CompletableFuture.completedFuture(
+          logResponse(InstallResponse.builder().withStatus(RaftResponse.Status.OK).build()));
     }
 
     if (!request.complete() && request.nextChunkId() == null) {
@@ -185,8 +189,8 @@ public class PassiveRole extends InactiveRole {
                   .build()));
     }
 
-    final var snapshotChunk = new SnapshotChunkImpl();
-    snapshotChunk.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
+    final SnapshotChunk snapshotChunk = decodeSnapshotChunk(request);
+    log.debug("Installing snapshot chunk {}", snapshotChunk);
 
     // If there is no pending snapshot, create a new snapshot.
     if (pendingSnapshot == null) {
@@ -221,7 +225,7 @@ public class PassiveRole extends InactiveRole {
                     .withStatus(RaftResponse.Status.ERROR)
                     .withError(
                         RaftError.Type.ILLEGAL_MEMBER_STATE,
-                        "Request chunk is was received out of order")
+                        "Request chunk was received out of order")
                     .build()));
       }
     }
@@ -383,6 +387,18 @@ public class PassiveRole extends InactiveRole {
                 .withError(
                     RaftError.Type.ILLEGAL_MEMBER_STATE, "Cannot request vote from RESERVE member")
                 .build()));
+  }
+
+  private SnapshotChunk decodeSnapshotChunk(final InstallRequest request) {
+    try {
+      final var snapshotChunkImpl = new SnapshotChunkImpl();
+      snapshotChunkImpl.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
+      return snapshotChunkImpl;
+    } catch (final RuntimeException e) {
+      // fallback for requests sent from pre 0.24.x brokers, where request.data() is the chunk
+      // contents; please remove once we do not support versions below 0.24.x
+      return LegacySnapshotChunk.ofInstallRequest(request);
+    }
   }
 
   private void abortPendingSnapshots() {

--- a/atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/FileBasedReceivedSnapshot.java
+++ b/atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/FileBasedReceivedSnapshot.java
@@ -38,7 +38,8 @@ import org.agrona.concurrent.UnsafeBuffer;
 import org.slf4j.Logger;
 
 public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
-
+  static final int TOTAL_COUNT_NULL_VALUE = SnapshotChunkDecoder.totalCountNullValue();
+  static final long SNAPSHOT_CHECKSUM_NULL_VALUE = SnapshotChunkDecoder.snapshotChecksumNullValue();
   private static final Logger LOGGER = new ZbLogger(FileBasedReceivedSnapshot.class);
   private static final boolean FAILED = false;
   private static final boolean SUCCESS = true;
@@ -58,8 +59,8 @@ public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
     this.metadata = metadata;
     this.snapshotStore = snapshotStore;
     this.directory = directory;
-    this.expectedSnapshotChecksum = Long.MIN_VALUE;
-    this.expectedTotalCount = Integer.MIN_VALUE;
+    this.expectedSnapshotChecksum = SNAPSHOT_CHECKSUM_NULL_VALUE;
+    this.expectedTotalCount = TOTAL_COUNT_NULL_VALUE;
   }
 
   @Override
@@ -79,6 +80,11 @@ public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
     }
 
     return expectedId.equals(chunkId);
+  }
+
+  @Override
+  public void setNextExpected(final ByteBuffer nextChunkId) {
+    expectedId = nextChunkId;
   }
 
   @Override
@@ -136,7 +142,7 @@ public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
   }
 
   private boolean isSnapshotChecksumInvalid(final long currentSnapshotChecksum) {
-    if (expectedSnapshotChecksum == Long.MIN_VALUE) {
+    if (expectedSnapshotChecksum == SNAPSHOT_CHECKSUM_NULL_VALUE) {
       this.expectedSnapshotChecksum = currentSnapshotChecksum;
     }
 
@@ -151,7 +157,7 @@ public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
   }
 
   private boolean isTotalCountInvalid(final int currentTotalCount) {
-    if (expectedTotalCount == Integer.MIN_VALUE) {
+    if (expectedTotalCount == TOTAL_COUNT_NULL_VALUE) {
       this.expectedTotalCount = currentTotalCount;
     }
 
@@ -178,47 +184,6 @@ public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
   }
 
   @Override
-  public void setNextExpected(final ByteBuffer nextChunkId) {
-    expectedId = nextChunkId;
-  }
-
-  @Override
-  public PersistedSnapshot persist() {
-    if (snapshotStore.hasSnapshotId(metadata.getSnapshotIdAsString())) {
-      abort();
-      return snapshotStore.getLatestSnapshot().orElseThrow();
-    }
-
-    final var files = directory.toFile().listFiles();
-    Objects.requireNonNull(files, "No chunks have been applied yet");
-
-    if (files.length != expectedTotalCount) {
-      throw new IllegalStateException(
-          String.format(
-              "Expected '%d' chunk files for this snapshot, but found '%d'. Files are: %s.",
-              expectedSnapshotChecksum, files.length, Arrays.toString(files)));
-    }
-
-    final var filePaths =
-        Arrays.stream(files).sorted().map(File::toPath).collect(Collectors.toList());
-    final long actualSnapshotChecksum;
-    try {
-      actualSnapshotChecksum = ChecksumUtil.createCombinedChecksum(filePaths);
-    } catch (final IOException e) {
-      throw new UncheckedIOException("Unexpected exception on calculating snapshot checksum.", e);
-    }
-
-    if (actualSnapshotChecksum != expectedSnapshotChecksum) {
-      throw new IllegalStateException(
-          String.format(
-              "Expected snapshot checksum %d, but calculated %d.",
-              expectedSnapshotChecksum, actualSnapshotChecksum));
-    }
-
-    return snapshotStore.newSnapshot(metadata, directory);
-  }
-
-  @Override
   public void abort() {
     try {
       LOGGER.debug("DELETE dir {}", directory);
@@ -231,6 +196,44 @@ public class FileBasedReceivedSnapshot implements ReceivedSnapshot {
     } catch (final IOException e) {
       LOGGER.warn("Failed to delete pending snapshot {}", this, e);
     }
+  }
+
+  @Override
+  public PersistedSnapshot persist() {
+    if (snapshotStore.hasSnapshotId(metadata.getSnapshotIdAsString())) {
+      abort();
+      return snapshotStore.getLatestSnapshot().orElseThrow();
+    }
+
+    final var files = directory.toFile().listFiles();
+    Objects.requireNonNull(files, "No chunks have been applied yet");
+
+    if (expectedTotalCount != TOTAL_COUNT_NULL_VALUE && files.length != expectedTotalCount) {
+      throw new IllegalStateException(
+          String.format(
+              "Expected '%d' chunk files for this snapshot, but found '%d'. Files are: %s.",
+              expectedSnapshotChecksum, files.length, Arrays.toString(files)));
+    }
+
+    if (expectedSnapshotChecksum != SNAPSHOT_CHECKSUM_NULL_VALUE) {
+      final var filePaths =
+          Arrays.stream(files).sorted().map(File::toPath).collect(Collectors.toList());
+      final long actualSnapshotChecksum;
+      try {
+        actualSnapshotChecksum = ChecksumUtil.createCombinedChecksum(filePaths);
+      } catch (final IOException e) {
+        throw new UncheckedIOException("Unexpected exception on calculating snapshot checksum.", e);
+      }
+
+      if (actualSnapshotChecksum != expectedSnapshotChecksum) {
+        throw new IllegalStateException(
+            String.format(
+                "Expected snapshot checksum %d, but calculated %d.",
+                expectedSnapshotChecksum, actualSnapshotChecksum));
+      }
+    }
+
+    return snapshotStore.newSnapshot(metadata, directory);
   }
 
   public Path getPath() {

--- a/atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/LegacySnapshotChunk.java
+++ b/atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/LegacySnapshotChunk.java
@@ -1,0 +1,122 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.snapshot.impl;
+
+import io.atomix.raft.protocol.InstallRequest;
+import io.atomix.raft.snapshot.SnapshotChunk;
+import io.atomix.utils.time.WallClockTimestamp;
+import io.zeebe.util.buffer.BufferUtil;
+import java.util.Arrays;
+import java.util.Objects;
+import org.agrona.concurrent.UnsafeBuffer;
+
+/**
+ * An implementation of {@link SnapshotChunk} which is compatible with the snapshot replication
+ * protocol of brokers pre 0.24.x. Should be removed once no versions below 0.24.x are supported.
+ */
+public final class LegacySnapshotChunk implements SnapshotChunk {
+  private final FileBasedSnapshotMetadata metadata;
+  private final String chunkId;
+  private final byte[] content;
+  private final long checksum;
+
+  private LegacySnapshotChunk(
+      final FileBasedSnapshotMetadata metadata,
+      final String chunkId,
+      final byte[] content,
+      final long checksum) {
+    this.metadata = metadata;
+    this.chunkId = chunkId;
+    this.content = content;
+    this.checksum = checksum;
+  }
+
+  public static LegacySnapshotChunk ofInstallRequest(final InstallRequest request) {
+    final var metadata =
+        new FileBasedSnapshotMetadata(
+            request.index(), request.term(), WallClockTimestamp.from(request.timestamp()));
+    final var chunkId = BufferUtil.bufferAsString(new UnsafeBuffer(request.chunkId()));
+    final var content = BufferUtil.bufferAsArray(new UnsafeBuffer(request.data()));
+    final var checksum = SnapshotChunkUtil.createChecksum(content);
+
+    return new LegacySnapshotChunk(metadata, chunkId, content, checksum);
+  }
+
+  @Override
+  public String getSnapshotId() {
+    return metadata.getSnapshotIdAsString();
+  }
+
+  @Override
+  public int getTotalCount() {
+    return FileBasedReceivedSnapshot.TOTAL_COUNT_NULL_VALUE;
+  }
+
+  @Override
+  public String getChunkName() {
+    return chunkId;
+  }
+
+  @Override
+  public long getChecksum() {
+    return checksum;
+  }
+
+  @Override
+  public byte[] getContent() {
+    return content;
+  }
+
+  @Override
+  public long getSnapshotChecksum() {
+    return FileBasedReceivedSnapshot.SNAPSHOT_CHECKSUM_NULL_VALUE;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    final LegacySnapshotChunk that = (LegacySnapshotChunk) o;
+    return getChecksum() == that.getChecksum()
+        && Objects.equals(metadata, that.metadata)
+        && Objects.equals(chunkId, that.chunkId)
+        && Arrays.equals(getContent(), that.getContent());
+  }
+
+  @Override
+  public int hashCode() {
+    int result = Objects.hash(metadata, chunkId, getChecksum());
+    result = 31 * result + Arrays.hashCode(getContent());
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "LegacySnapshotChunk{"
+        + "metadata="
+        + metadata
+        + ", chunkId='"
+        + chunkId
+        + '\''
+        + ", checksum="
+        + checksum
+        + '}';
+  }
+}

--- a/atomix/cluster/src/test/java/io/atomix/raft/snapshot/impl/LegacySnapshotChunkTest.java
+++ b/atomix/cluster/src/test/java/io/atomix/raft/snapshot/impl/LegacySnapshotChunkTest.java
@@ -1,0 +1,122 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.snapshot.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.raft.protocol.InstallRequest;
+import java.nio.ByteBuffer;
+import org.junit.Test;
+
+public class LegacySnapshotChunkTest {
+  @Test
+  public void shouldHaveCorrectChunkName() {
+    // given
+    final var request =
+        newInstallRequest(1, 1, System.currentTimeMillis(), "foo", "baz".getBytes());
+
+    // when
+    final var chunk = LegacySnapshotChunk.ofInstallRequest(request);
+
+    // then
+    assertThat(chunk.getChunkName()).isEqualTo("foo");
+  }
+
+  @Test
+  public void shouldHaveCorrectSnapshotId() {
+    // given
+    final var request = newInstallRequest(2, 3, 10, "baz", "baz".getBytes());
+
+    // when
+    final var chunk = LegacySnapshotChunk.ofInstallRequest(request);
+
+    // then
+    assertThat(chunk.getSnapshotId()).isEqualTo("2-3-10");
+  }
+
+  @Test
+  public void shouldHaveNullTotalCount() {
+    // given
+    final var request = newInstallRequest(2, 3, 10, "baz", "baz".getBytes());
+
+    // when
+    final var chunk = LegacySnapshotChunk.ofInstallRequest(request);
+
+    // then
+    assertThat(chunk.getTotalCount()).isEqualTo(FileBasedReceivedSnapshot.TOTAL_COUNT_NULL_VALUE);
+  }
+
+  @Test
+  public void shouldHaveNullSnapshotChecksum() {
+    // given
+    final var request = newInstallRequest(2, 3, 10, "foo", "baz".getBytes());
+
+    // when
+    final var chunk = LegacySnapshotChunk.ofInstallRequest(request);
+
+    // then
+    assertThat(chunk.getSnapshotChecksum())
+        .isEqualTo(FileBasedReceivedSnapshot.SNAPSHOT_CHECKSUM_NULL_VALUE);
+  }
+
+  @Test
+  public void shouldHaveCorrectChecksum() {
+    // given
+    final byte[] content = "baz".getBytes();
+    final long checksum = SnapshotChunkUtil.createChecksum(content);
+    final var request = newInstallRequest(2, 3, 10, "foo", content);
+
+    // when
+    final var chunk = LegacySnapshotChunk.ofInstallRequest(request);
+
+    // then
+    assertThat(chunk.getChecksum()).isEqualTo(checksum);
+  }
+
+  @Test
+  public void shouldHaveCorrectContent() {
+    // given
+    final byte[] content = "baz".getBytes();
+    final var request = newInstallRequest(2, 3, 10, "foo", content);
+
+    // when
+    final var chunk = LegacySnapshotChunk.ofInstallRequest(request);
+
+    // then
+    assertThat(chunk.getContent()).isEqualTo(content);
+  }
+
+  private InstallRequest newInstallRequest(
+      final long index,
+      final int term,
+      final long timestamp,
+      final String chunkName,
+      final byte[] content) {
+    return new InstallRequest(
+        1,
+        MemberId.anonymous(),
+        index,
+        term,
+        timestamp,
+        1,
+        ByteBuffer.wrap(chunkName.getBytes()),
+        null,
+        ByteBuffer.wrap(content),
+        true,
+        true);
+  }
+}

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -78,7 +78,7 @@
     <version.asm>8.0.1</version.asm>
     <version.testcontainers>1.14.3</version.testcontainers>
     <version.netflix.concurrency>0.3.6</version.netflix.concurrency>
-    <version.zeebe-test-container>0.32.0</version.zeebe-test-container>
+    <version.zeebe-test-container>0.33.0</version.zeebe-test-container>
     <version.feel-scala>1.11.2</version.feel-scala>
     <version.restassert>4.3.0</version.restassert>
     <version.spring-framework>5.2.7.RELEASE</version.spring-framework>

--- a/test-util/src/main/java/io/zeebe/test/util/asserts/TopologyAssert.java
+++ b/test-util/src/main/java/io/zeebe/test/util/asserts/TopologyAssert.java
@@ -10,6 +10,7 @@ package io.zeebe.test.util.asserts;
 import io.zeebe.client.api.response.BrokerInfo;
 import io.zeebe.client.api.response.Topology;
 import java.util.List;
+import java.util.function.Consumer;
 import java.util.stream.Collectors;
 import org.assertj.core.api.AbstractAssert;
 
@@ -19,7 +20,7 @@ public class TopologyAssert extends AbstractAssert<TopologyAssert, Topology> {
     super(topology, TopologyAssert.class);
   }
 
-  public static TopologyAssert assertThat(Topology actual) {
+  public static TopologyAssert assertThat(final Topology actual) {
     return new TopologyAssert(actual);
   }
 
@@ -42,6 +43,29 @@ public class TopologyAssert extends AbstractAssert<TopologyAssert, Topology> {
           "Expected <%s> partitions at each broker, but found brokers with different partition count <%s>",
           partitionCount, brokersWithUnexpectedPartitionCount);
     }
+
+    return this;
+  }
+
+  public final TopologyAssert doesNotContainBroker(final int nodeId) {
+    isNotNull();
+
+    final List<Integer> brokers =
+        actual.getBrokers().stream().map(BrokerInfo::getNodeId).collect(Collectors.toList());
+    if (brokers.contains(nodeId)) {
+      failWithMessage(
+          "Expected topology not to contain broker with ID %d, but found the following: [%s]",
+          nodeId, brokers);
+    }
+
+    return this;
+  }
+
+  public final TopologyAssert hasBrokerSatisfying(final Consumer<BrokerInfo> condition) {
+    isNotNull();
+
+    final List<BrokerInfo> brokers = actual.getBrokers();
+    newListAssertInstance(brokers).anySatisfy(condition);
 
     return this;
   }

--- a/upgrade-tests/pom.xml
+++ b/upgrade-tests/pom.xml
@@ -61,6 +61,11 @@
       <groupId>net.jodah</groupId>
       <artifactId>failsafe</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.awaitility</groupId>
+      <artifactId>awaitility</artifactId>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
 
   <build>

--- a/upgrade-tests/src/test/java/io/zeebe/test/ContainerStateRule.java
+++ b/upgrade-tests/src/test/java/io/zeebe/test/ContainerStateRule.java
@@ -98,7 +98,8 @@ class ContainerStateRule extends TestWatcher {
             .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERNAME", "zeebe-cluster")
             .withNetwork(network)
             .withEmbeddedGateway(gatewayVersion == null)
-            .withLogLevel(Level.DEBUG);
+            .withLogLevel(Level.DEBUG)
+            .withDebug(true);
 
     Failsafe.with(CONTAINER_START_RETRY_POLICY).run(() -> broker.start());
 

--- a/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java
+++ b/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java
@@ -125,7 +125,7 @@ public class RollingUpdateTest {
   }
 
   @Test
-  public void shouldReplicateSnapshotAcrossVersions() throws InterruptedException {
+  public void shouldReplicateSnapshotAcrossVersions() {
     // given
     Startables.deepStart(containers).join();
 
@@ -156,6 +156,7 @@ public class RollingUpdateTest {
           .pollInterval(Duration.ofMillis(100))
           .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, brokerId));
 
+      // need to create enough records to allow for the creation of a snapshot
       for (int i = 0; i < 100; i++) {
         Awaitility.await("process instance creation")
             .atMost(Duration.ofSeconds(5))
@@ -180,6 +181,7 @@ public class RollingUpdateTest {
           .untilAsserted(() -> assertTopologyContainsUpgradedBroker(client, brokerId));
     }
 
+    // then
     assertBrokerHasAtLeastOneSnapshot(1);
   }
 

--- a/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java
+++ b/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java
@@ -7,30 +7,67 @@
  */
 package io.zeebe.test;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.client.ZeebeClient;
+import io.zeebe.client.api.response.WorkflowInstanceEvent;
+import io.zeebe.client.api.worker.JobHandler;
 import io.zeebe.containers.ZeebeBrokerContainer;
 import io.zeebe.containers.ZeebePort;
-import io.zeebe.test.util.AutoCloseableRule;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.test.util.asserts.TopologyAssert;
 import io.zeebe.util.VersionUtil;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.time.Duration;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
+import org.agrona.IoUtil;
+import org.awaitility.Awaitility;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.junit.rules.TemporaryFolder;
 import org.slf4j.event.Level;
 import org.testcontainers.containers.Network;
+import org.testcontainers.lifecycle.Startable;
 import org.testcontainers.lifecycle.Startables;
 
 public class RollingUpdateTest {
-  private static final Logger LOG = LoggerFactory.getLogger(ContainerStateRule.class);
   private static final String OLD_VERSION = VersionUtil.getPreviousVersion();
-  private static final String CURRENT_VERSION = "current-test";
+  private static final String NEW_VERSION = VersionUtil.getVersion();
+  private static final String IMAGE_TAG = "current-test";
+  private static final BpmnModelInstance PROCESS =
+      Bpmn.createExecutableProcess("process")
+          .startEvent()
+          .serviceTask("task1", s -> s.zeebeJobType("firstTask"))
+          .serviceTask("task2", s -> s.zeebeJobType("secondTask"))
+          .endEvent()
+          .done();
 
-  @Rule public AutoCloseableRule autoCloseable = new AutoCloseableRule();
+  private static final File SHARED_DATA;
+
+  static {
+    final var sharedDataPath =
+        Optional.ofNullable(System.getenv("ZEEBE_CI_SHARED_DATA"))
+            .map(Paths::get)
+            .orElse(Paths.get(System.getProperty("tmpdir", "/tmp"), "shared"));
+    SHARED_DATA = sharedDataPath.toAbsolutePath().toFile();
+    IoUtil.ensureDirectoryExists(SHARED_DATA, "temporary folder for Docker");
+  }
+
+  @Rule public TemporaryFolder tmpFolder = new TemporaryFolder(SHARED_DATA);
 
   private List<ZeebeBrokerContainer> containers;
   private String initialContactPoints;
@@ -47,47 +84,230 @@ public class RollingUpdateTest {
 
     containers =
         Arrays.asList(
-            manageClosable(new ZeebeBrokerContainer(OLD_VERSION)),
-            manageClosable(new ZeebeBrokerContainer(OLD_VERSION)),
-            manageClosable(new ZeebeBrokerContainer(OLD_VERSION)));
+            new ZeebeBrokerContainer(OLD_VERSION),
+            new ZeebeBrokerContainer(OLD_VERSION),
+            new ZeebeBrokerContainer(OLD_VERSION));
 
     configureBrokerContainer(0, containers);
     configureBrokerContainer(1, containers);
     configureBrokerContainer(2, containers);
   }
 
-  @Test
-  public void shouldBeAbleToRestartContainerWithSameVersion() {
-    // given
-    final var index = 0;
-    final var sameVersion = OLD_VERSION;
-    Startables.deepStart(containers).join();
-    containers.get(index).shutdownGracefully(Duration.ofSeconds(30));
-
-    // when
-    final var zeebeBrokerContainer = replaceBrokerContainer(index, sameVersion);
-
-    // then
-    zeebeBrokerContainer.start();
+  @After
+  public void tearDown() {
+    containers.parallelStream().forEach(Startable::stop);
   }
 
   @Test
   public void shouldBeAbleToRestartContainerWithNewVersion() {
     // given
     final var index = 0;
-    final var newVersion = CURRENT_VERSION;
     Startables.deepStart(containers).join();
     containers.get(index).shutdownGracefully(Duration.ofSeconds(30));
 
     // when
-    final var zeebeBrokerContainer = replaceBrokerContainer(index, newVersion);
+    final var zeebeBrokerContainer = upgradeBroker(index);
 
     // then
-    zeebeBrokerContainer.start();
+    try (final var client = newZeebeClient(containers.get(1))) {
+      Awaitility.await()
+          .atMost(Duration.ofSeconds(5))
+          .pollInterval(Duration.ofMillis(100))
+          .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, index));
+
+      zeebeBrokerContainer.start();
+
+      Awaitility.await()
+          .atMost(Duration.ofSeconds(5))
+          .pollInterval(Duration.ofMillis(100))
+          .untilAsserted(() -> assertTopologyContainsUpgradedBroker(client, index));
+    }
   }
 
-  private ZeebeBrokerContainer replaceBrokerContainer(final int index, final String newVersion) {
-    final var broker = new ZeebeBrokerContainer(newVersion);
+  @Test
+  public void shouldReplicateSnapshotAcrossVersions() throws InterruptedException {
+    // given
+    Startables.deepStart(containers).join();
+
+    // when
+    final var availableBroker = containers.get(0);
+    try (final var client = newZeebeClient(availableBroker)) {
+      deployProcess(client);
+
+      // potentially retry in case we're faster than the deployment distribution
+      Awaitility.await("process instance creation")
+          .atMost(Duration.ofSeconds(5))
+          .pollInterval(Duration.ofMillis(100))
+          .ignoreExceptions()
+          .until(() -> createWorkflowInstance(client), Objects::nonNull)
+          .getWorkflowInstanceKey();
+    }
+
+    try (final var client = newZeebeClient(availableBroker)) {
+      final var brokerId = 1;
+      var container = containers.get(brokerId);
+
+      container.shutdownGracefully(Duration.ofSeconds(30));
+
+      // until previous version points to 0.24, we cannot yet tune failure detection to be fast,
+      // so wait long enough for the broker to be removed even in slower systems
+      Awaitility.await("broker is removed from topology")
+          .atMost(Duration.ofSeconds(20))
+          .pollInterval(Duration.ofMillis(100))
+          .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, brokerId));
+
+      for (int i = 0; i < 100; i++) {
+        Awaitility.await("process instance creation")
+            .atMost(Duration.ofSeconds(5))
+            .pollInterval(Duration.ofMillis(100))
+            .ignoreExceptions()
+            .until(() -> createWorkflowInstance(client), Objects::nonNull)
+            .getWorkflowInstanceKey();
+      }
+
+      // wait for a snapshot - even if 0 is not the leader, it will get the replicated snapshot
+      // which is a good indicator we now have a snapshot
+      Awaitility.await("broker 0 has created a snapshot")
+          .atMost(Duration.ofMinutes(2)) // twice the snapshot period
+          .pollInterval(Duration.ofMillis(500))
+          .untilAsserted(() -> assertBrokerHasAtLeastOneSnapshot(0));
+
+      container = upgradeBroker(brokerId);
+      container.start();
+      Awaitility.await("upgraded broker is added to topology")
+          .atMost(Duration.ofSeconds(10))
+          .pollInterval(Duration.ofMillis(100))
+          .untilAsserted(() -> assertTopologyContainsUpgradedBroker(client, brokerId));
+    }
+
+    assertBrokerHasAtLeastOneSnapshot(1);
+  }
+
+  @Test
+  public void shouldPerformRollingUpgrade() {
+    // given
+    Startables.deepStart(containers).join();
+
+    // when
+    final long firstWorkflowInstanceKey;
+    var availableBroker = containers.get(0);
+    try (final var client = newZeebeClient(availableBroker)) {
+      deployProcess(client);
+
+      // potentially retry in case we're faster than the deployment distribution
+      firstWorkflowInstanceKey =
+          Awaitility.await("process instance creation")
+              .atMost(Duration.ofSeconds(5))
+              .pollInterval(Duration.ofMillis(100))
+              .ignoreExceptions()
+              .until(() -> createWorkflowInstance(client), Objects::nonNull)
+              .getWorkflowInstanceKey();
+    }
+
+    for (int i = containers.size() - 1; i >= 0; i--) {
+      try (final var client = newZeebeClient(availableBroker)) {
+        final var brokerId = i;
+        var container = containers.get(i);
+
+        container.shutdownGracefully(Duration.ofSeconds(30));
+
+        // until previous version points to 0.24, we cannot yet tune failure detection to be fast,
+        // so wait long enough for the broker to be removed even in slower systems
+        Awaitility.await("broker is removed from topology")
+            .atMost(Duration.ofSeconds(20))
+            .pollInterval(Duration.ofMillis(100))
+            .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, brokerId));
+
+        container = upgradeBroker(i);
+        container.start();
+        Awaitility.await("upgraded broker is added to topology")
+            .atMost(Duration.ofSeconds(10))
+            .pollInterval(Duration.ofMillis(100))
+            .untilAsserted(() -> assertTopologyContainsUpgradedBroker(client, brokerId));
+
+        availableBroker = container;
+      }
+    }
+
+    // then
+    final Map<Long, List<String>> activatedJobs = new HashMap<>();
+    final var expectedOrderedJobs = List.of("firstTask", "secondTask");
+    final JobHandler jobHandler =
+        (jobClient, job) -> {
+          jobClient.newCompleteCommand(job.getKey()).send().join();
+          activatedJobs.compute(
+              job.getWorkflowInstanceKey(),
+              (ignored, list) -> {
+                final var appendedList =
+                    Optional.ofNullable(list).orElse(new CopyOnWriteArrayList<>());
+                appendedList.add(job.getType());
+                return appendedList;
+              });
+        };
+
+    try (final var client = newZeebeClient(availableBroker)) {
+      final var secondWorkflowInstanceKey = createWorkflowInstance(client).getWorkflowInstanceKey();
+      final var expectedActivatedJobs =
+          Map.of(
+              firstWorkflowInstanceKey,
+              expectedOrderedJobs,
+              secondWorkflowInstanceKey,
+              expectedOrderedJobs);
+      client.newWorker().jobType("firstTask").handler(jobHandler).open();
+      client.newWorker().jobType("secondTask").handler(jobHandler).open();
+
+      Awaitility.await("all jobs have been activated")
+          .atMost(Duration.ofSeconds(5))
+          .untilAsserted(() -> assertThat(activatedJobs).isEqualTo(expectedActivatedJobs));
+    }
+  }
+
+  private WorkflowInstanceEvent createWorkflowInstance(final ZeebeClient client) {
+    return client
+        .newCreateInstanceCommand()
+        .bpmnProcessId("process")
+        .latestVersion()
+        .variables(Map.of("foo", "bar"))
+        .send()
+        .join();
+  }
+
+  private void deployProcess(final ZeebeClient client) {
+    client
+        .newDeployCommand()
+        .addWorkflowModel(PROCESS, "process.bpmn")
+        .send()
+        .join(10, TimeUnit.SECONDS);
+  }
+
+  private void assertTopologyContainsUpgradedBroker(
+      final ZeebeClient zeebeClient, final int brokerId) {
+    final var topology = zeebeClient.newTopologyRequest().send().join();
+    TopologyAssert.assertThat(topology)
+        .isComplete(containers.size(), 1)
+        .hasBrokerSatisfying(
+            brokerInfo -> {
+              assertThat(brokerInfo.getNodeId()).isEqualTo(brokerId);
+              assertThat(brokerInfo.getVersion()).isEqualTo(NEW_VERSION);
+            });
+  }
+
+  private void assertTopologyDoesNotContainerBroker(final ZeebeClient client, final int brokerId) {
+    final var topology = client.newTopologyRequest().send().join();
+    TopologyAssert.assertThat(topology)
+        .doesNotContainBroker(brokerId)
+        .isComplete(containers.size() - 1, 1);
+  }
+
+  private ZeebeClient newZeebeClient(final ZeebeBrokerContainer container) {
+    return ZeebeClient.newClientBuilder()
+        .usePlaintext()
+        .brokerContactPoint(container.getExternalAddress(ZeebePort.GATEWAY))
+        .build();
+  }
+
+  private ZeebeBrokerContainer upgradeBroker(final int index) {
+    final var broker = new ZeebeBrokerContainer(IMAGE_TAG);
     containers.set(index, broker);
     return configureBrokerContainer(index, containers);
   }
@@ -97,26 +317,39 @@ public class RollingUpdateTest {
     final int clusterSize = brokers.size();
     final var broker = brokers.get(index);
     final var hostName = "broker-" + index;
+    final var volumePath = getBrokerVolumePath(index);
     broker.withNetworkAliases(hostName);
 
+    // once old version is 0.24 and more membership configuration is exposed, further tune for fast
+    // failure detection
     return broker
         .withNetwork(network)
         .withEnv("ZEEBE_BROKER_NETWORK_HOST", "0.0.0.0")
         .withEnv("ZEEBE_BROKER_NETWORK_ADVERTISED_HOST", hostName)
         .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERNAME", "zeebe-cluster")
-        .withEnv("ZEEBE_BROKER_DATA_SNAPSHOTPERIOD", "1m")
-        .withEnv("ZEEBE_BROKER_DATA_LOGSEGMENTSIZE", "1MB")
-        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "1MB")
+        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "128KB")
         .withEnv("ZEEBE_BROKER_CLUSTER_NODEID", String.valueOf(index))
         .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERSIZE", String.valueOf(clusterSize))
         .withEnv("ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR", String.valueOf(clusterSize))
         .withEnv("ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", initialContactPoints)
-        .withLogLevel(Level.DEBUG)
-        .withDebug(false);
+        .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPFAILURETIMEOUT", "5000")
+        .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPINTERVAL", "100")
+        .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPPROBEINTERVAL", "100")
+        .withEnv("ZEEBE_BROKER_DATA_SNAPSHOTPERIOD", "1m")
+        .withFileSystemBind(volumePath.toString(), "/usr/local/zeebe/data")
+        .withLogLevel(Level.DEBUG);
   }
 
-  private <T extends AutoCloseable> T manageClosable(final T closeable) {
-    autoCloseable.manage(closeable);
-    return closeable;
+  private Path getBrokerVolumePath(final int index) {
+    final var file = new File(tmpFolder.getRoot(), "broker-" + index);
+    IoUtil.ensureDirectoryExists(file, "broker shared data folder");
+
+    return file.toPath().toAbsolutePath();
+  }
+
+  private void assertBrokerHasAtLeastOneSnapshot(final int index) {
+    final var path = getBrokerVolumePath(index);
+    final var snapshotPath = path.resolve("raft-partition/partitions/1/snapshots");
+    assertThat(snapshotPath).isNotEmptyDirectory();
   }
 }

